IntegrationTestDS3/IntegrationTestDS3Client.cs

```diff
@@ -0,0 +1,382 @@
+       {
+           // delete it
+           DeleteFolderRequest request = new Ds3.Calls.DeleteFolderRequest(TESTBUCKET, FOLDER);
+           _client.DeleteFolder(request);
+       }
+       **/
+       [Test]
+       [ExpectedException(typeof(Ds3.Runtime.Ds3BadStatusCodeException))]
+       public void Test0520GetBadBucket()
+       {
+           var request = new Ds3.Calls.GetBucketRequest("NoBucket" + DateTime.Now.Ticks);
+           _client.GetBucket(request);
+       }
+
+       [Test]
+       public void Test0910DeleteObject()
+       {
+           var antefolder = listBucketObjects();
+           int antefoldercount = antefolder.Count();
+           // delete the first book 
+           string book = BOOKS.First<string>();
+           var request = new Ds3.Calls.DeleteObjectRequest(TESTBUCKET, string.Empty + book);
+           _client.DeleteObject(request);
+
+           // one less ?
+           var postfolder = listBucketObjects();
+           int postfoldercount = postfolder.Count();
+           Assert.AreEqual(antefoldercount - postfoldercount, 1);
+       }
+
+       [Test]
+       [ExpectedException(typeof(Ds3.Runtime.Ds3BadStatusCodeException))]
+       public void Test0915DeleteDeletedObject()
+       {
+           // delete the first book, again 
+           string book = BOOKS.First<string>();
+           var request = new Ds3.Calls.DeleteObjectRequest(TESTBUCKET, string.Empty + book);
+               _client.DeleteObject(request);
+       }
+
+       [Test]
+       public void Test0920DeleteObjectWithPrefix()
+       {
+           var antefolder = listBucketObjects();
+           int antefoldercount = antefolder.Count();
+           // delete the dirst book 
+           string book = BOOKS.First<string>();
+           var request = new Ds3.Calls.DeleteObjectRequest(TESTBUCKET, PREFIX + book);
+           _client.DeleteObject(request);
+
+           // one less ?
+           var postfolder = listBucketObjects();
+           int postfoldercount = postfolder.Count();
+           Assert.AreEqual(antefoldercount - postfoldercount, 1);
+       }
+
+       [Test] 
+       public void Test0990CleanUp()
+        {
+            var items = _helpers.ListObjects(TESTBUCKET);
+
+            // Loop through all of the objects in the bucket.
+            foreach (var obj in items)
+            {
+                DeleteObject(TESTBUCKET, obj.Name);
+            }
+            DeleteBucket(TESTBUCKET);
+        }
+
+       public void DeleteObject(string bucketname, string objectName)
+       {
+           var request = new Ds3.Calls.DeleteObjectRequest(bucketname, objectName);
+           _client.DeleteObject(request);
+       }
+
+       public void DeleteBucket(string bucketname)
+       {
+           var request = new Ds3.Calls.DeleteBucketRequest(bucketname);
+           _client.DeleteBucket(request);
+       }
+
+
+       #endregion sequential tests
+
+        #region main
+       public static void Main()
```

Comment:
Not needed per se, but it is very handy to be able to run the tests in the IDE. Your call.

jk

From: Ryan Moore [mailto:notifications@github.com]
Sent: Tuesday, October 06, 2015 9:36 AM
To: SpectraLogic/ds3_net_sdk <ds3_net_sdk@noreply.github.com>
Cc: John Kranz <johnk@spectralogic.com>
Subject: Re: [ds3_net_sdk] Add Integration tests (NETSDK-44) (#56)


In IntegrationTestDS3/IntegrationTestDS3Client.cs<https://github.com/SpectraLogic/ds3_net_sdk/pull/56#discussion_r41280449>:

> +       {

> +           var request = new Ds3.Calls.DeleteObjectRequest(bucketname, objectName);

> +           _client.DeleteObject(request);

> +       }

> +

> +       public void DeleteBucket(string bucketname)

> +       {

> +           var request = new Ds3.Calls.DeleteBucketRequest(bucketname);

> +           _client.DeleteBucket(request);

> +       }

> +

> +

> +       #endregion sequential tests

> +

> +        #region main

> +       public static void Main()

Do we need a main? I thought that nunit did all of this for you.

—
Reply to this email directly or view it on GitHub<https://github.com/SpectraLogic/ds3_net_sdk/pull/56/files#r41280449>.